service/history/workflow/update/state.go

```diff
@@ -59,6 +65,31 @@ func (s state) String() string {
 	return "unrecognized state"
 }
 
+// LifecycleStage maps the states of the Update state machine to the values of
+// enumspb.UpdateWorkflowExecutionLifecycleStage.
+func (s state) LifecycleStage() (enumspb.UpdateWorkflowExecutionLifecycleStage, error) {
+	switch s {
+	case stateAdmitted:
+		return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED, nil
+	case stateProvisionallyRequested:
+		return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED, nil
+	case stateRequested:
+		return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED, nil
+	case stateProvisionallyAccepted:
+		return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED, nil
+	case stateAccepted:
+		return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ACCEPTED, nil
+	case stateProvisionallyCompleted:
+		// Transition could be due to either validation rejection or completion
+		// of Update, so Admitted is the most advanced stage known to have been
+		// reached.
+		return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED, nil
+	case stateCompleted:
+		return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED, nil
+	}
+	return enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_UNSPECIFIED, fmt.Errorf("unrecognized state: %v", s)
```

Comment:
I would ignore this lint error, we have too many predefined errors already in this codebase.